graylog2-server/src/main/java/org/graylog2/dashboards/DashboardServiceImpl.java

```diff
@@ -107,20 +110,19 @@ public Dashboard load(String id) throws NotFoundException {
 
     @Override
     public void updateWidgetPositions(Dashboard dashboard, WidgetPositionsRequest positions) throws ValidationException {
-        Map<String, Map<String, Object>> map = Maps.newHashMap();
+        final List<WidgetPosition> widgetPositions = new ArrayList<>();
```

Comment:
We can use the size of `positions.positions()` to allocate the correct size of the array right from the start.